pkg/kubelet/kubelet_node_status.go

```diff
@@ -160,6 +167,13 @@ func (kl *Kubelet) reconcileExtendedResource(initialNode, node *v1.Node) bool {
 		if v1helper.IsExtendedResourceName(k) {
 			node.Status.Capacity[k] = *resource.NewQuantity(int64(0), resource.DecimalSI)
 			node.Status.Allocatable[k] = *resource.NewQuantity(int64(0), resource.DecimalSI)
+			if initialNode.Status.Capacity == nil {
```

Comment:
This new logic is confusing. The old logic made sense - if an existing node object has extended resources, set them to zero in the new object. On that note, it would be great if you can rename the method and the variables to better reflect the function logic. `resetExtendedResourceCapacityAndAllocatable(newNode, existingNode *v1.Node)`